mongo_orchestration/common.py

```diff
@@ -80,19 +84,36 @@ def _strip_auth(self, proc_params):
         params.pop("clusterAuthMode", None)
         return params
 
-    def mongodb_auth_uri(self, hosts):
-        """Get a connection string with all info necessary to authenticate."""
-        parts = ['mongodb://']
+    def mongodb_uri(self, hosts, uri_opts):
+        """Append URI options for auth, TLS, etc."""
+        """Append TLS options"""
+        if self.ssl_params:
+            ssl_params = self.ssl_params
+            ssl_params.update(DEFAULT_SSL_OPTIONS)
+            """Rewrite ssl* option names to tls*"""
+            for sslKey, tlsKey in SSL_TO_TLS_OPTION_MAPPINGS.items():
+                sslValue = ssl_params.pop(sslKey)
+                if sslValue:
+                    if isinstance(sslValue, bool):
+                        sslValue = json.dumps(sslValue)
+                    uri_opts.append(tlsKey + '=' + sslValue)
```

Comment:
I think we need to uri encode the sslValue here. Especially for the ssl_certfile and sslCAFile file path:
```
if sys.version_info[0] == 2:
    from urllib import quote_plus
else:
    from urllib.parse import quote_plus
    
    
...
uri_opts.append(tlsKey + '=' + quote_plus(sslValue))
```

Reply:
Ah, yes. Applied as suggested.